testing/python/quantize/test_tilelang_quantization.py

```
@@ -0,0 +1,54 @@
+    ],
+)
+@tilelang.testing.requires_cuda
+def test_u32_to_int_to_float_sign_extends_subword_values(nbit, values):
+    """Signed sub-word decode should preserve negative values from uint32 storage."""
+
+    lanes_per_word = 32 // nbit
+    num_words = math.ceil(len(values) / lanes_per_word)
+    num_values = len(values)
+
+    @T.prim_func
+    def main(I: T.Tensor((num_words,), "uint32"), O: T.Tensor((num_values,), "float32")):
```

Ruff (0.15.20)

[error] Ambiguous variable name: `I`

(E741)